gomock/call.go

```diff
@@ -435,12 +435,43 @@ func (c *Call) call() []func([]any) []any {
 }
 
 // InOrder declares that the given calls should occur in order.
-func InOrder(calls ...*Call) {
+// It panics if the type of any of the arguments isn't *Call or a generated
+// mock with an embedded *Call.
+func InOrder(args ...any) {
+	var calls []*Call
+	for i := 0; i < len(args); i++ {
+		if call := getCall(args[i]); call != nil {
+			calls = append(calls, call)
+			continue
+		}
+		panic("InOrder parameters type must be either a *Call or a generated mock with an embedded *Call")
```

Comment:
Can we also print as part of the error message additional details on what was passed through that caused the error?